Teams docs (#32949)
demisto/content: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md

Note: in step 5, if you choose **Use existing app registration**, make sure to delete the previous created bot with the same app id, remove it from the team it was added to as well.  

### Grant the Demisto Bot Permissions in Microsoft Graph

> In order to connect to the Azure Network Security Groups use one of the following methods:

Reviewer: ```suggestion
In order to connect to Microsoft Teams use one of the following authentication methods:
```
*my mistake

1. *Client Credentials Flow*
2. *Authorization Code Flow*

##### Client Credentials Flow

> Note: [The chat commands](#chat-commands) are only supported when using the `Authorization Cod flow`.

Reviewer: ```suggestion
Note: [The chat commands](#chat-commands) are only supported when using the `Authorization Code flow`.
```

    | Disable Automatic Notifications | Whether to disable automatic notifications to the configured notifications channel. | False |
    | Allow external users to create incidents via direct message |  | False |
    | The header of an external form hyperlink. |  | False |

>     | Trust any certificate (not secure) | Do not check for Cortex XSOAR version 8 and XSIAM | False |

Reviewer 2: ```suggestion
    | Trust any certificate (not secure) | Do not check for Cortex XSOAR version 8 and Cortex XSIAM. | False |
```

    | Use system proxy settings |  | False |
    | Long running instance |  | True |
    | Listen port, e.g., 7000 (Required for investigation mirroring and direct messages) | longRunningPort | False |
    | Incident type | Incident type. | False |

4. Click **Test** to validate the URLs, token, and connection.
5. Click the **Save & exit** button.

#### Configuring the instance with the chosen authentication flow

##### Authentication Using the Client Credentials Flow

> 1. Choose the 'Client Credentials' option in the **Authentication Type** parameter.
> 2. Enter your Client/Application ID in the **Bot ID** parameter. 
> 3. Enter your Client Secret in the **Bot Password** parameter.

Reviewer 2: ```suggestion
1. Choose the 'Client Credentials' option in the *Authentication Type* parameter.
2. Enter your Client/Application ID in the *Bot ID* parameter. 
3. Enter your Client Secret in the *Bot Password* parameter.
```

4. Save the instance.

##### Authentication Using the Authorization Code Flow

> 1. Choose the 'Authorization Code' option in the **Authentication Type** parameter.
> 2. Enter your Client/Application ID in the **Bot ID** parameter. 
> 3. Enter your Client Secret in the **Bot Password** parameter.
> 4. Enter your Application redirect URI in the **Application redirect URI** parameter.
> 5. Run the !microsoft-teams-generate-login-url command in the War Room and follow the instructions.

Reviewer 2: ```suggestion
1. Choose the 'Authorization Code' option in the *Authentication Type* parameter.
2. Enter your Client/Application ID in the *Bot ID* parameter. 
3. Enter your Client Secret in the *Bot Password* parameter.
4. Enter your Application redirect URI in the *Application redirect URI* parameter.
5. Run the ***!microsoft-teams-generate-login-url*** command in the War Room and follow the instructions.
```

6. Save the instance.
7. Run the ***!microsoft-teams-auth-test*** command. A 'Success' message should be printed to the War Room.

### Add the Demisto Bot to a Team

> - Note: the following need to be done after configuring the integration on Cortex XSOAR/XSIAM (the previous step).

Reviewer 2: ```suggestion
- Note: The following needs to be done after configuring the integration on Cortex XSOAR/Cortex XSIAM (the previous step).
```

4. Click **Test** to validate the URLs, token, and connection.
5. Click the **Save & exit** button.

#### Configuring the instance with the chosen authentication flow

##### Authentication Using the Client Credentials Flow

1. Choose the 'Client Credentials' option in the **Authentication Type** parameter.
2. Enter your Client/Application ID in the **Bot ID** parameter. 
3. Enter your Client Secret in the **Bot Password** parameter.
4. Save the instance.

##### Authentication Using the Authorization Code Flow

1. Choose the 'Authorization Code' option in the **Authentication Type** parameter.
2. Enter your Client/Application ID in the **Bot ID** parameter. 
3. Enter your Client Secret in the **Bot Password** parameter.
4. Enter your Application redirect URI in the **Application redirect URI** parameter.

> 5. Copy the following URL and replace the **TENANT_ID**, **CLIENT_ID** and **REDIRECT_URI** with your own client ID and redirect URI, accordingly.

Reviewer: ```suggestion
5. Run the !microsoft-teams-generate-login-url command in the War Room and follow the instructions.
```

> https://login.microsoftonline.com/TENANT_ID/oauth2/v2.0/authorize?response_type=code&response_mode=query&scope=offline_access%20https%3A%2F%2Fgraph.microsoft.com%2F.default&client_id=CLIENT_ID&redirect_uri=REDIRECT_URI&state=12345. When prompted, accept the Microsoft authorization request for the required permissions. You will be automatically redirected to a link with the following structure:

Reviewer: ```suggestion
```

> ```REDIRECT_URI?code=AUTH_CODE&state=12345&session_state=SESSION_STATE```

Reviewer: ```suggestion
```

> 6. Copy the **AUTH_CODE** (without the “code=” prefix) and paste it in your instance configuration under the **Authorization code** parameter. 

Reviewer: ```suggestion
```

> 7. Save the instance.

Reviewer: ```suggestion
6. Save the instance.
```

> 8. Run the ***!microsoft-teams-auth-test*** command. A 'Success' message should be printed to the War Room.

Reviewer: ```suggestion
7. Run the ***!microsoft-teams-auth-test*** command. A 'Success' message should be printed to the War Room.
```

### mirror-investigation
***

> Mirrors the Cortex XSOAR/XSIAM investigation to the specified Microsoft Teams channel. Supports only standard channels.

Reviewer 2: ```suggestion
Mirrors the Cortex XSOAR/Cortex XSIAM investigation to the specified Microsoft Teams channel. Supports only standard channels.
```

> **Note**: Mirrored channels could be used to run Cortex XSOAR/XSIAM built-in commands.

Reviewer 2: ```suggestion
**Note**: Mirrored channels could be used to run Cortex XSOAR/Cortex XSIAM built-in commands.
```

>2. Copy the `AUTH_CODE` (without the `code=` prefix, and the `session_state` parameter)
>and paste it in your instance configuration under the **Authorization code** parameter.
>

## Running commands from Microsoft Teams

> You can run Cortex XSOAR/XSIAM commands, according to the user permissions, from Microsoft Teams in a mirrored investigation channel.

Reviewer 2: ```suggestion
You can run Cortex XSOAR/Cortex XSIAM commands, according to the user permissions, from Microsoft Teams in a mirrored investigation channel.
```

        - From the Cortex XSOAR machine to localhost.
          - Note: The web server supports only POST method queries. 
        
   3. If the cURL queries were sent successfully, you should see the following line in Cortex XSOAR logs: `Finished processing Microsoft Teams activity successfully`.

>    4. If you're working with secured communication (HTTPS), make sure that you provided a valid certificate. (Not XSOAR/XSIAM Rerouting ).

Reviewer 2: ```suggestion
   4. If you're working with secured communication (HTTPS), make sure that you provided a valid certificate. (Not for Cortex XSOAR/Cortex XSIAM Rerouting ).
```

       1. Run `openssl s_client -connect <domain.com>:443` .
       2. Verify that the returned value of the `Verify return code` field is `0 (ok)`, otherwise, it's not a valid certificate.
    

>    5. Try inserting your configured message endpoint in a browser and click **Enter**. If `Method Not Allowed` is returned, the endpoint is valid and ready to communicate, otherwise, it needs to be handled according to the returned error's message. (Not for XSOAR 8 OR XSIAM).

Reviewer 2: ```suggestion
   5. Try inserting your configured message endpoint in a browser and click **Enter**. If `Method Not Allowed` is returned, the endpoint is valid and ready to communicate, otherwise, it needs to be handled according to the returned error's message. (Not for Cortex XSOAR 8 OR Cortex XSIAM).
```
